### PR TITLE
[FIXED JENKINS-36048] Properly choose a credentials provider when appropriate

### PR DESCRIPTION
**Pull request by abayer**
https://issues.jenkins-ci.org/browse/JENKINS-36048

The lack of ids on the relevant `<span>` means I had to use a fairly demented xpath to find it. Buuuuut now it works!

cc @reviewbybees esp @stephenc @kwhetstone

### CODE REVIEW

**Comment by andresrc** [comment]
:bee:

**Comment by abayer** [comment]
Ok, fixed it in the no-additional-providers use case and made it no longer need a hardcoded yui ID. Sorry about that!

File: src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java [hunk 1]
```diff
@@ -30,7 +31,24 @@ public SshSlaveLauncher(PageObject context, String path) {
...
+            self().findElement(by.xpath("//button[@class='credentials-add-menu']")).click();
+
+            for (WebElement menuItem : self().findElements(by.xpath(providerXpathExpr))) {
+                if (menuItem.isDisplayed()) {
+                    menuItem.click();
+                    break;
+                }
+            }
+        } else {
+            self().findElement(by.button("Add")).click();
```
Comment (kwhetstone) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java]: Will this eventually be removed?  I assume it's here for an older version of ssh-slaves.

File: src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java [hunk 2]
```diff
@@ -30,7 +31,24 @@ public SshSlaveLauncher(PageObject context, String path) {
...
+            String providerXpathExpr = "//div[contains(@class,'credentials-add-menu-items')]"
+                    + "/div[@class='bd']/ul[@class='first-of-type']/li[contains(@class, 'yuimenuitem')]"
+                    + "/span[contains(@class,'yuimenuitemlabel') and contains(text(), 'Jenkins')]";
+
+            self().findElement(by.xpath("//button[@class='credentials-add-menu']")).click();
+
+            for (WebElement menuItem : self().findElements(by.xpath(providerXpathExpr))) {
+                if (menuItem.isDisplayed()) {
+                    menuItem.click();
+                    break;
```
Comment (kwhetstone) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java]: If the ordering in the menu changes are we going to have to update the test?

**Comment by kwhetstone** [comment]
:bee: Added some questions.  Looks good!

File: src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java (hunk 1 above)
Comment (abayer) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java]: Nope - which version of ssh-slaves is irrelevant! The question is entirely whether multiple applicable credentials providers are installed.

File: src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java (hunk 2 above)
Comment (abayer) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java]: Ordering shouldn't matter - that's why we're looking for the `"Jenkins"` menu item, not just the first one.

**Comment by andresrc** [comment]
:bee:

**Comment by abayer** [comment]
@reviewbybees done

**Comment by abayer** [comment]
@olivergondza If need be, feel free to revert this, but I'm merging it. =)

**Pull request merged**